File: clients/java/src/main/java/org/aigen/oabp/model/SubmitRequest.java
package org.aigen.oabp.model;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The request body for {@code POST /missions/{id}/submit}.
 *
 * @param submitterAgentId the id of the agent submitting the deliverable (required)
 * @param proof            the deliverable proof — free text or a URL (required)
 */
public record SubmitRequest(
        @JsonProperty("submitter_agent_id") String submitterAgentId,
        @JsonProperty("proof") String proof) {

    public SubmitRequest {
        if (submitterAgentId == null || submitterAgentId.isBlank()) {
            throw new IllegalArgumentException("submitterAgentId is required and must be non-blank");
        }
        if (proof == null || proof.isBlank()) {
            throw new IllegalArgumentException("proof is required and must be non-blank");
        }
    }
}

File: clients/kotlin/src/main/kotlin/org/aigen/oabp/model/SubmitRequest.kt
package org.aigen.oabp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * The request body for `POST /missions/{id}/submit`.
 *
 * @property submitterAgentId the id of the agent submitting the deliverable (required, non-blank).
 * @property proof the deliverable proof — free text or a URL (required, non-blank).
 */
@Serializable
public data class SubmitRequest(
    @SerialName("submitter_agent_id")
    val submitterAgentId: String,
    val proof: String,
) {
    init {
        require(submitterAgentId.isNotBlank()) { "submitterAgentId is required and must be non-blank" }
        require(proof.isNotBlank()) { "proof is required and must be non-blank" }
    }
}
